dbus/dbus.hpp

```diff
@@ -26,29 +27,41 @@ class SDBusPlus
      * @param[in] bus - bus.
      * @param[in] busName - bus name.
      * @param[in] objPath - object path.
-     * @param[in] value - value to be set.
+     * @param[in] value - value to be set, in degrees, regardless of unitMilli.
+     * @param[in] unitMilli - true to set millidegrees, false to set degrees.
      */
-    static auto
+    static void
         setValueProperty(sdbusplus::bus::bus& bus, const std::string& busName,
-            const std::string& objPath, const int64_t& value)
+            const std::string& objPath, const double& value, bool unitMilli)
```

Comment:
Thanks, but the code already uses double in other places, so it's not a new addition here. This program already requires the kernel to do correct FPU emulation on ARM, but that's unchanged from how it was before. Passing in a double preserves accuracy, as it can then be converted "just in time" to integer at the last moment, after all other processing is done.

Reply:
Yep, I just wonder if double is a bad choice from the beginning in all other components. :/

Reply:
About the only disadvantage I would see is performance (since floating-point requires software emulation to work on ARM processor). It seems fine to me, because the main loop only runs once per second. CPU load is 4% to 7% on read-margin-temp and I suspect most of that is D-Bus overhead, not because of double. It would be worth benchmarking, though.